inst/pages/exercises.qmd

```diff
@@ -1741,6 +1741,188 @@ pheatmap(
     annotation_col = as.data.frame(sample_metadata))
 ```
 
+### *Bonus* Regression Charts to Visualize Microbiome Data :
+
+Regression analysis is a powerful statistical method that can help examine
+the relationship between two or more variables. In microbiome research, 
+regression charts help in visualizing how changes in microbial communities
+correlate with various environmental factors, health outcomes, or other
+biological variables.
+
+Let us do a regression analysis exercise using the `GlobalPatterns` dataset
+with the following variables:
+
+1. ShannonIndex
+2. Body Mass Index (BMI) (bmi_group, variable in the GlobalPatterns dataset)
+
+Steps:
+
+1. Import the dataset into R; here we are using a demo dataset.
+2. Create a scatter plot with "Shannon Index" on the y-axis and "BMI_group" 
+on the x-axis.
+3. Fit a linear regression model to the data.  In R, you can use `ggplot2` for 
+plotting and `lm()` function for linear modeling.
+4. Add the regression line to the scatter plot for visualization.
+5. Examine the slope, intercept, and R-squared value to 
+understand the relationship between the variables.
+6. Test the linear modeling assumptions.
+7. Treat other possible covariates.
+8. Perform visualizations and interpret the analysis outcomes. You can use
+visualization methods from packages like miaViz, scater, etc.
+
+```{r}
+#| label: extra-regressionchart
+#| code-fold: true
+#| code-summary: "Show solution"
+#| eval: false
+
+# Load the necessary libraries
+library(ggplot2)
+library(TreeSummarizedExperiment)
+library(mia)
+library(miaViz)
+library(scater)
+
+# Load the demo dataset 
+data("GlobalPatterns", package = "mia")
+tse <- GlobalPatterns
+
+# Calculate the Shannon diversity index
+diversity <- estimateDiversity(tse, assay.type = "counts", index = "shannon")
+
+# Extract the Shannon diversity index and bmi_group
+shannon_index <- diversity$shannon
+bmi_group <- colData(tse)$SampleType  # Using SampleType as a proxy for BMI_Group
+
+# Create the data frame
+data <- data.frame(
+  Shannon_Index = shannon_index,
+  BMI_Group = bmi_group
+)
+
+# Perform the regression analysis
+model <- lm(Shannon_Index ~ BMI_Group, data = data)
+
+# Create the plot with the regression line
+p <- ggplot(data, aes(x = BMI_Group, y = Shannon_Index)) +
+  geom_point(color = 'blue') +
+  geom_smooth(method = 'lm', color = 'red') +
```

Comment:
geom_smooth() creates its own linear model, you do not need the previous lm() command for this ggplot visualization. It would be good to remove either one of these, or explain in the text that geom_smooth() recalculates the lm for visualization